### PR TITLE
fixed search algorithms

### PR DESCRIPTION
**Pull request by Miraddo**
I am working on making more cleaner the codes and tests of searching algorithms.

- [x] binary search | fixed grammar and make the test case more readable.
- [x] interpolation search  | fixed grammar and make the test case more readable.
- [x] linear search  | fixed grammar and make the test case more readable.
- [x] jump search  | fixed grammar and make the test case more readable.

### CODE REVIEW

File: topics/go/algorithms/searches/binarysearch/binarysearch.go [hunk 1]
```diff
@@ -45,34 +45,35 @@ func binarySearchIterative(sortedList []int, target int) (int, error) {
...
+	case leftIdx > rightIdx:
+		return -1, fmt.Errorf("target not found")
 
-	// 	If List[mid] is bigger than Target
-	//	I returned the binarySearchRecursive function with changing the high value to mid minus by one
-	if List[mid] > Target {
-		return binarySearchRecursive(List, Target, low, mid-1)
-	}
+	// Check if we found the target.
+	case sortedList[midIdx] == target:
```
Comment (ardan-bkennedy) [inline topics/go/algorithms/searches/binarysearch/binarysearch.go:61]: Start with the assignment like above.

File: topics/go/algorithms/searches/binarysearch/binarysearch.go [hunk 2]
```diff
@@ -45,34 +45,35 @@ func binarySearchIterative(sortedList []int, target int) (int, error) {
...
-	for low > high {
-		return -1, fmt.Errorf("sorry value not found")
-	}
+	switch {
 
-	// 	check is List[mid] value equal with Target value than return the middle value
-	if List[mid] == Target {
-		return mid, nil
-	}
+	// check if the left index is greater than right index,
```
Comment (ardan-bkennedy) [inline topics/go/algorithms/searches/binarysearch/binarysearch.go:55]: This comment needs to start with a capital C.   Check if...

File: topics/go/algorithms/searches/binarysearch/binarysearch.go [hunk 3]
```diff
@@ -45,34 +45,35 @@ func binarySearchIterative(sortedList []int, target int) (int, error) {
...
+	// Then recall itself.
+	case sortedList[midIdx] > target:
+		return binarySearchRecursive(sortedList, target, leftIdx, midIdx-1)
+
+	// If the value is less than the target, cut the list
+	// by moving the leftIdx into the list.
+	// Then recall itself.
+	case sortedList[midIdx] < target:
+		return binarySearchRecursive(sortedList, target, midIdx+1, rightIdx)
 	}
```
Comment (ardan-bkennedy) [inline topics/go/algorithms/searches/binarysearch/binarysearch.go:76]: There are two spaces here. I don't think you are running gofmt. What editor are you using?

File: topics/go/algorithms/searches/binarysearch/binarysearch.go (hunk 1 above)
Comment (Miraddo) [reply topics/go/algorithms/searches/binarysearch/binarysearch.go:61]: I used `VScode`, sometimes `GoLand` or `VIM`. 

I configured `gofmt` and `staticcheck` for all of them.

I checked it directly.

gofmt -w .\binarysearch.go
gofmt -w .\binarysearch_test.go

I guess now it's work

File: topics/go/algorithms/searches/binarysearch/binarysearch.go (hunk 3 above)
Comment (Miraddo) [reply topics/go/algorithms/searches/binarysearch/binarysearch.go:76]: I used `VScode`, sometimes `GoLand` or `VIM`. 

I configured `gofmt` and `staticcheck` for all of them.

I checked it directly.

gofmt -w .\binarysearch.go
gofmt -w .\binarysearch_test.go

I guess now it's work

File: topics/go/algorithms/searches/binarysearch/binarysearch.go (hunk 1 above)
Comment (Miraddo) [reply topics/go/algorithms/searches/binarysearch/binarysearch.go]: I changed the comments for the recursive part as well.

File: topics/go/algorithms/searches/binarysearch/binarysearch.go (hunk 2 above)
Comment (Miraddo) [reply topics/go/algorithms/searches/binarysearch/binarysearch.go]: I fixed it

**Comment by ardan-bkennedy** [comment]
In jumpsearch.go I see another opportunity to use a switch / case

File: topics/go/algorithms/searches/interpolationsearch/interpolationsearch.go [hunk 4]
```diff
@@ -1,61 +1,73 @@
...
-		return -1
-	}
+	// Check until we find the target or searched the list.
+	if leftIdx <= rightIdx && target >= sortedList[leftIdx] && target <= sortedList[rightIdx] && len(sortedList) > 0 {
 
-	if low <= high && target >= list[low] && target <= list[high] {
+		positionIdx := leftIdx + int(float64(rightIdx-leftIdx)/float64(sortedList[rightIdx]-sortedList[leftIdx]))*(target-sortedList[leftIdx])
 
-		position := low + int(float64(high-low)/float64(list[high]-list[low]))*(target-list[low])
+		value := sortedList[positionIdx]
```
Comment (ardan-bkennedy) [inline topics/go/algorithms/searches/interpolationsearch/interpolationsearch.go:51]: Missing same comments from function above. Be consistent.

File: topics/go/algorithms/searches/interpolationsearch/interpolationsearch.go [hunk 5]
```diff
@@ -1,61 +1,73 @@
...
 
-		if list[position] == target {
-			return position
-		}
+		switch {
+		// Check if we found the target.
```
Comment (ardan-bkennedy) [inline topics/go/algorithms/searches/interpolationsearch/interpolationsearch.go:54]: Missing line space above the comment.

File: topics/go/algorithms/searches/jumpsearch/jumpsearch.go [hunk 6]
```diff
@@ -1,64 +1,66 @@
...
-	// if index is smaller or equal with the length of array continue the loop
-	for index <= len(list)-1 {
+	// Loop until we find the target or break the loop if target is smaller than the sortedList value.
+	for index <= len(sortedList)-1 {
 
-		// if the value of list and target is equal,
-		// the index of array will be returned
-		if list[index] == target {
+		// Check if we found the target.
+		if sortedList[index] == target {
```
Comment (ardan-bkennedy) [inline topics/go/algorithms/searches/jumpsearch/jumpsearch.go:27]: Why not use a case here like we are using everywhere else?

**Comment by ardan-bkennedy** [comment]
I never know how people review code like this. I will accept this and send you new comments.

**Comment by ardan-bkennedy** [comment]
Look at jump and linear search. You have opportunities to use switch / case.

Also notice comments don't go past the 80 column line.

Inside of interpolation I did the following.

```
a := int(float64(rightIdx-leftIdx) / float64(sortedList[rightIdx]-sortedList[leftIdx]))
b := target - sortedList[leftIdx]
positionIdx := leftIdx + a*b
```

Can we be smarter about what `a` and `b` are? Maybe we can break `a` into more parts?

**Comment by Miraddo** [comment]
> Look at jump and linear search. You have opportunities to use switch / case.
> 
> Also notice comments don't go past the 80 column line.
> 
> Inside of interpolation I did the following.
> 
> ```
> a := int(float64(rightIdx-leftIdx) / float64(sortedList[rightIdx]-sortedList[leftIdx]))
> b := target - sortedList[leftIdx]
> positionIdx := leftIdx + a*b
> ```
> 
> Can we be smarter about what `a` and `b` are? Maybe we can break `a` into more parts?

Could we have something like this?

```
a := float64(rightIdx-leftIdx)
b := float64(sortedList[rightIdx]-sortedList[leftIdx])
c := int(a / b)
d := target - sortedList[leftIdx]
positionIdx := leftIdx + c*d

```

**Comment by ardan-bkennedy** [comment]
Works for me. It would be nice to know what you are doing and why.

**Pull request merged**